dashboard/app/models/user.rb

```diff
@@ -1663,16 +1663,58 @@ def assigned_course_or_script?
   # Example: true when the primary_script is being used for a TopCourse on /home
   # @return [Array{CourseData, ScriptData}] an array of hashes of script and
   # course data
-  def recent_courses_and_scripts(exclude_primary_script)
-    primary_script_id = Queries::ScriptActivity.primary_script(self).try(:id)
+  def recent_pl_courses_and_units(exclude_primary_script)
+    primary_script_id = Queries::ScriptActivity.primary_pl_unit(self).try(:id)
 
     # Filter out user_scripts that are already covered by a course
-    unit_group_units_script_ids = courses_as_student.map(&:default_unit_group_units).flatten.pluck(:script_id).uniq
+    unit_group_units_script_ids = courses_as_participant.map(&:default_unit_group_units).flatten.pluck(:script_id).uniq
 
     user_scripts = Queries::ScriptActivity.in_progress_and_completed_scripts(self).
       select {|user_script| !unit_group_units_script_ids.include?(user_script.script_id)}
 
-    user_script_data = user_scripts.map do |user_script|
+    pl_user_scripts = user_scripts.select {|us| us.script.pl_course?}
+
+    user_script_data = pl_user_scripts.map do |user_script|
+      # Skip this script if we are excluding the primary script and this is the
+      # primary script.
+      if exclude_primary_script && user_script[:script_id] == primary_script_id
+        nil
+      else
+        script_id = user_script[:script_id]
+        script = Script.get_from_cache(script_id)
+        {
+          name: script[:name],
+          title: data_t_suffix('script.name', script[:name], 'title'),
+          description: data_t_suffix('script.name', script[:name], 'description_short', default: ''),
+          link: script_path(script),
+        }
+      end
+    end.compact
+
+    user_course_data = courses_as_participant.select(&:pl_course?).map(&:summarize_short)
+
+    user_course_data + user_script_data
+  end
+
+  # Return a collection of courses and scripts for the user.
+  # First in the list will be courses enrolled in by the user's sections.
+  # Following that will be all scripts in which the user has made progress that # are not in any of the enrolled courses.
+  # @param exclude_primary_script [boolean]
+  # Example: true when the primary_script is being used for a TopCourse on /home
+  # @return [Array{CourseData, ScriptData}] an array of hashes of script and
+  # course data
+  def recent_student_courses_and_units(exclude_primary_script)
```

Comment:
just to check my understanding: the term `student` in these methods is essentially referring to "non-pl" right? The term feels confusing to me but I can't think of a better one

Reply:
Yes thats right it means non pl but also the only things that are non-pl are student facing courses. I also couldn't think of something better but agree its not the easiest term.